Sources/BraintreeThreeDSecure/BTThreeDSecureV2Provider.swift

```diff
@@ -122,14 +122,20 @@ extension BTThreeDSecureV2Provider: CardinalValidationDelegate {
                 completion: completionHandler
             )
         case .error, .timeout:
-            let userInfo = [NSLocalizedDescriptionKey: validateResponse.errorDescription]
+            var userInfo = [NSLocalizedDescriptionKey: validateResponse.errorDescription]
             var errorCode: Int = BTThreeDSecureError.unknown.errorCode
 
             if validateResponse.errorNumber == 1050 {
                 errorCode = BTThreeDSecureError.failedAuthentication("").errorCode
+                completionHandler(nil, BTThreeDSecureError.failedAuthentication(""))
+            } else if validateResponse.actionCode == .timeout {
+                errorCode = BTThreeDSecureError.exceededTimeoutLimit.errorCode
+                userInfo = [NSLocalizedDescriptionKey: BTThreeDSecureError.exceededTimeoutLimit.localizedDescription]
+                completionHandler(nil, BTThreeDSecureError.exceededTimeoutLimit)
+            } else {
+                completionHandler(nil, BTThreeDSecureError.exceededTimeoutLimit)
```

Comment:
I don't think we want our catch all else block to throw the timeout error

Reply:
I think also we can on line 124 break out the `case .error, .timeout:` into 2 separate cases where the new timeout case can throw the new error you added.

Reply:
That's a good idea, makes it cleaner